### DISCUSSION TITLE
Session 05.10.2020

### DISCUSSION THREAD

**Discussion by hnformentin** [equinor/tdd_workshop#31] [General]
Context
- First time workshop run with more than 15 people
- Two teams
- Diversity in levels of experience (beginner to advanced)
- Fully virtual
- 3.5 hours, three parts split by a 15 min and a 5 min break

Concepts discussed (topics that came up when sharing experience):
- Coverage (only unit tests?)
- Time of tests (difference between unit tests and integration tests - every time is run, overnight run?)
- Consistency of tests over time using README.md to explain how the team approaches test
- Use of remotes (cleanness but also security - when working with secrets)
- Level of tests, more difficult to refactor when too many tests? Tests coupled with implementation?
- How to deal when expectation is different than the plan of the workshop?

Feedback/points for improvement

Preparation:
- Very useful to have the prep-chat, aligning on some expectations

First part:
- Sharing the screen during the introduction round was not a great idea - the video gets very laggy. 
- Picking people using specific terms and asking them to define those terms, that's great! 
- The red green cycle could be explained better. Maybe even using a simple live example
- Very good both for content and interaction to have a more experienced person invited
- Identify `specialists` in specific areas to bring them to answer advanced questions during the whole workshop (not necessarily to be online all the time, but to be available to join). In this case would be: testing in the context of cloud infrastructure, large data sets, data sciences (ML pipelines), advanced pytest (fixtures, decorators, mocking), databricks and cluster

Second part:
- The participants were very well prepared with the pre-requisites
- Reinforce that the practical part is expected to be done with RGR cycle
- Advise to read carefully the problem, very often people are misled by the simplicity
- Group people considering availability/experience (?)
- Be attentive when participants ask for help in the main channel (maybe more than one facilitator/instructor would help)

Third part:
- At that stage, we were less energetic. The challenge (what will be done next in the projects) should probably be in a different format, for example working as peers and bringing up suggestions
- A simple feedback form to be replied during the workshop would be essential for large groups

**Comment by pale-otter** [comment]
I really liked the workshop, which was well prepared from the facilitator. It gave a lot of food for thought, as the topics above indicate. 

I may have been better to have a smaller group for the workshop, as this may have made it easier to have more focused discussions. On the other hand; more people present -> more experiences to be shared (potentially more eye-openers).

**Reply by vslatten** [reply]
I think the workshop was great. If I were to do it again, I would perhaps want a whole day workshop, so we could dive into more complex topics. On the other hand, I think many of us joined partly because "it's only half a day".

Quick nitpick: The date in the conversations title should be "5.10.2021"
